test/integration/authz_v2_test.go

```diff
@@ -0,0 +1,399 @@
+		if err != nil {
+			return false, err
+		}
+		if len(authPolicyList) != 0 {
+			return false, nil
+		}
+		return true, nil
+	})
+	assert.Nil(t, err, "Authorization policy list not empty")
+}
+
+// Basic test case
+// 1. Athenz Domain exists, created service, checked for Authorization Policy
+// Initial: No services, existing AD
+// Input Actions: create service with annotation
+// Output: AP created
+func TestCreateAuthorizationPolicy(t *testing.T) {
+	e := fixtures.GetBasicRbacV2Case(nil)
+	rolloutAndValidateAuthorizationPolicyScenario(t, e, create, create)
+	cleanupAuthorizationRbac(t, e)
+}
+
+// Manual update AP
+// Initial: Existing service, AP, AD
+// Input Actions: edit AP resource with rule addition/deletion
+// Output: AP updated to match with existing AD
+func TestUpdatedAuthorizationPolicyRestoresOriginal(t *testing.T) {
+	// Basic setup
+	// Creates Athenz Domain, Service verifies Authorization Policy
+	e := fixtures.GetBasicRbacV2Case(nil)
+	rolloutAndValidateAuthorizationPolicyScenario(t, e, create, create)
+
+	// Retrieve a modified authorization policy
+	modifiedResources := fixtures.GetBasicRbacV2Case(&fixtures.RbacV2Modifications{
+		ModifyAthenzDomain: []func(signedDomain *zms.SignedDomain){},
+		ModifyServices:     [][]func(service *v1.Service){},
+		ModifyAuthorizationPolicies: [][]func(policy *v1beta1.AuthorizationPolicy){
+			[]func(policy *v1beta1.AuthorizationPolicy){
+				func(policy *v1beta1.AuthorizationPolicy) {
+					policy.Action = v1beta1.AuthorizationPolicy_DENY
+				},
+			},
+		},
+	})
+
+	// Update the Authorization policy with modified value
+	ap := modifiedResources.AuthorizationPolicies[0]
+	authorizationPolicy := framework.Global.IstioClientset.Get(ap.GroupVersionKind(), ap.Name, ap.Namespace)
+	assert.NotNil(t, authorizationPolicy, "Already set authorization policy cannot be nil")
+	updatedValue := authorizationPolicy.DeepCopy()
+	updatedValue.Spec = ap.Spec
+	_, err := framework.Global.IstioClientset.Update(updatedValue)
+	assert.Nil(t, err, "Update to Authorization policy must not fail")
+
+	// Check that the original authorization policy is restored.
+	rolloutAndValidateAuthorizationPolicyScenario(t, e, noop, noop)
+	cleanupAuthorizationRbac(t, e)
+}
+
+// Manual delete AP
+// Initial: Existing service, AP, AD
+// Input Actions: delete AP resource
+// Output: AP recreated to match with existing AD
+func TestDeleteAuthorizationPolicyRestoresOriginal(t *testing.T) {
+	// Initial set up
+	e := fixtures.GetBasicRbacV2Case(nil)
+	rolloutAndValidateAuthorizationPolicyScenario(t, e, create, create)
+
+	// Delete the Authorization policy created
+	ap := e.AuthorizationPolicies[0]
+	err := framework.Global.IstioClientset.Delete(ap.GroupVersionKind(), ap.Name, ap.Namespace)
+	assert.Nil(t, err, "Delete of authorization policy should not fail")
+
+	// Verification and cleanup
+	rolloutAndValidateAuthorizationPolicyScenario(t, e, noop, noop)
+	cleanupAuthorizationRbac(t, e)
+}
+
+// Update AP
```

Comment:
Can you rename this test case to update AD?

Reply:
Done, updated the comment .